src/main/groovy/edu/oregonstate/mist/api/Resource.groovy

```diff
@@ -83,4 +105,90 @@ abstract class Resource {
                 details: properties.get('internalServerError.details')
         ))
     }
+
+    void setEndpointUri(URI endpointUri) {
+        this.endpointUri = endpointUri
+    }
+
+    /**
+     * Returns string url to use in pagination links.
+     *
+     * @param params
+     * @return
+     */
+    protected String getPaginationUrl(def params) {
+        def uriAndPath = endpointUri.toString() + uriInfo.getPath()
+        def nonNullParams = params.clone()
+        // convert pageVariable to page[variable]
+        nonNullParams["page[number]"] = nonNullParams['pageNumber']
+        nonNullParams["page[size]"] = nonNullParams['pageSize']
+        nonNullParams.remove('pageSize')
+        nonNullParams.remove('pageNumber')
+
+        // remove empty GET parameters
+        nonNullParams = nonNullParams.findAll { it.value } .collect { k, v -> "$k=$v" }
```

Comment:
the names and values should be urlencoded. does java not have a class for constructing urls?